### PR TITLE
mlego fix product id and sync oled code

### PR DESCRIPTION
**Pull request by alinelena**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mlego/m48/rev1/config.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 /* USB Device descriptor parameter */
 #define PRODUCT_ID 0x0001
-#define DEVICE_VER 0x0001
+#define DEVICE_VER 0x6261
```
Comment (zvecr) [inline keyboards/mlego/m48/rev1/config.h:22]:
```
#define PRODUCT_ID 0x6261
#define DEVICE_VER 0x0001
```

File: keyboards/mlego/m60/rev1/config.h [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 /* USB Device descriptor parameter */
 #define PRODUCT_ID 0x0001
-#define DEVICE_VER 0x0001
+#define DEVICE_VER 0x6161
```
Comment (zvecr) [inline keyboards/mlego/m60/rev1/config.h:22]:
```
#define PRODUCT_ID 0x6161
#define DEVICE_VER 0x0001
```

**Review by zvecr — COMMENTED** [review]
Something like this? (as DEVICE_VER no longer looks like a bcdUSB)

File: keyboards/mlego/m48/rev1/config.h (hunk 1 above)
Comment (alinelena) [reply keyboards/mlego/m48/rev1/config.h]: yes was mistake fixed now... noticed when testing via

File: keyboards/mlego/m65/keymaps/default/keymap.c [hunk 3]
```diff
@@ -24,6 +24,17 @@ enum layer_names {
     _ADJ
 };
 
+#ifdef OLED_ENABLE
```
Comment (drashna) [inline keyboards/mlego/m65/keymaps/default/keymap.c:27]: If anything, the oled code here, especially since it appears to be the same for each keymap, should be moved to the keyboard's c file (m65.c, or rev#.c) instead of the keymap c file.  And switched to using something like this:
```c
#ifdef OLED_ENABLE
bool oled_task_kb(void) {
    if (!oled_task_user()) { 
        return false; 
    }
    // oled code
    return true;
}
#endif
```

This allows the keyboard to have a default behavior for the board, that will also work on the qmk configurator, and will allow for users to customize the behavior, if desired.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m65/m65.c [hunk 4]
```diff
@@ -16,3 +16,183 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+            if (clockwise) {
+                tap_code_delay(KC_VOLD, MEDIA_KEY_DELAY);
+            } else {
+                tap_code_delay(KC_VOLU, MEDIA_KEY_DELAY);
+            }
+        }
+    }
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/mlego/m65/m65.c:125]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
```

File: keyboards/mlego/m65/m65.c [hunk 5]
```diff
@@ -16,3 +16,183 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+        for (uint8_t j = 0; j < OLED_DISPLAY_WIDTH; ++j) {
+          oled_write_raw_byte(0x0, i*OLED_DISPLAY_WIDTH + j);
+        }
+      }
+      clear_logo = false;
+    }
+}
+
+#    define SHOW_LOGO 5000
+bool oled_task_user(void) {
```
Comment (drashna) [inline keyboards/mlego/m65/m65.c:188]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) { return false; }
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m65/rev4/rules.mk [hunk 6]
```diff
@@ -2,7 +2,8 @@
 MCU = STM32F401
 
 # Bootloader selection
-BOOTLOADER = stm32-dfu
+BOOTLOADER=tinyuf2
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev4/rules.mk:5]:
```suggestion
BOOTLOADER = tinyuf2
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/mlego/m48/rev1/config.h [hunk 7]
```diff
@@ -56,3 +56,6 @@
 #define RGBLIGHT_EFFECT_RGB_TEST
 #define RGBLIGHT_EFFECT_ALTERNATING
 #define RGBLIGHT_EFFECT_TWINKLE
+
+// you want to comment this if using stm32-dfu as bootloader
+#define FEE_PAGE_BASE_ADDRESS 0x0800c000
```
Comment (drashna) [inline keyboards/mlego/m48/rev1/config.h:61]: I'm not sure that this works? 

Have you confirmed that it actually does?

Comment (alinelena) [reply keyboards/mlego/m48/rev1/config.h:61]: you are right is not even compiling once I activate via...

**Comment by drashna** [comment]
Just to make sure, you've verified that the config works and saves settings properly?

File: keyboards/mlego/m48/rev1/config.h (hunk 7 above)
Comment (alinelena) [reply keyboards/mlego/m48/rev1/config.h:61]: ok fixed and confirmed with dasky's changes custom ld.

File: quantum/via.c [hunk 8]
```diff
@@ -72,7 +72,7 @@ void eeconfig_update_rgb_matrix(void);
 // Can be called in an overriding via_init_kb() to test if keyboard level code usage of
 // EEPROM is invalid and use/save defaults.
 bool via_eeprom_is_valid(void) {
-    char *  p      = QMK_BUILDDATE; // e.g. "2019-11-05-11:29:54"
+    static const char p[20]      = QMK_BUILDDATE; // e.g. "2019-11-05-11:29:54"
```
Comment (drashna) [inline quantum/via.c:75]: These changes need to be reverted and opened as a separate PR.
